dotnet/src/Connectors/Connectors.Memory.Qdrant/QdrantMemoryStore.cs

```diff
@@ -0,0 +1,369 @@
+/// The embedding data persists between subsequent instances and has similarity search capability.
+/// </remarks>
+public class QdrantMemoryStore : IMemoryStore
+{
+    /// <summary>
+    /// Constructor for a memory store backed by a Qdrant Vector database instance.
+    /// </summary>
+    /// <param name="host"></param>
+    /// <param name="port"></param>
+    /// <param name="vectorSize"></param>
+    /// <param name="logger"></param>
+    public QdrantMemoryStore(string host, int port, int vectorSize, ILogger? logger = null)
+    {
+        this._qdrantClient = new QdrantVectorDbClient(endpoint: host, port: port, vectorSize: vectorSize, log: logger);
+    }
+
+    /// <summary>
+    /// Constructor for a memory store backed by a <see cref="IQdrantVectorDbClient"/>
+    /// </summary>
+    public QdrantMemoryStore(IQdrantVectorDbClient client)
+    {
+        this._qdrantClient = client;
+    }
+
+    /// <inheritdoc/>
+    public async Task CreateCollectionAsync(string collectionName, CancellationToken cancel = default)
+    {
+        if (!await this._qdrantClient.DoesCollectionExistAsync(collectionName, cancel: cancel))
+        {
+            await this._qdrantClient.CreateCollectionAsync(collectionName, cancel: cancel);
+        }
+    }
+
+    /// <inheritdoc/>
+    public async Task<bool> DoesCollectionExistAsync(string collectionName, CancellationToken cancel = default)
+    {
+        return await this._qdrantClient.DoesCollectionExistAsync(collectionName, cancel: cancel);
+    }
+
+    /// <inheritdoc/>
+    public IAsyncEnumerable<string> GetCollectionsAsync(CancellationToken cancel = default)
+    {
+        return this._qdrantClient.ListCollectionsAsync(cancel: cancel);
+    }
+
+    /// <inheritdoc/>
+    public async Task DeleteCollectionAsync(string collectionName, CancellationToken cancel = default)
+    {
+        if (!await this._qdrantClient.DoesCollectionExistAsync(collectionName, cancel: cancel))
+        {
+            await this._qdrantClient.DeleteCollectionAsync(collectionName, cancel: cancel);
+        }
+    }
+
+    public async Task<string> UpsertAsync(string collectionName, MemoryRecord record, CancellationToken cancel = default)
+    {
+        var vectorData = await this.ConvertFromMemoryRecordAsync(collectionName, record, cancel);
+
+        if (vectorData == null)
+        {
+            throw new VectorDbException(VectorDbException.ErrorCodes.FailedToConvertMemoryRecordToQdrantVectorRecord, $"Failed to convert MemoryRecord to QdrantVectorRecord");
+        }
+
+        try
+        {
+            await this._qdrantClient.UpsertVectorsAsync(
+                collectionName,
+                new[] { vectorData },
+                cancel: cancel);
+        }
+        catch (HttpRequestException e)
+        {
+            throw new VectorDbException(VectorDbException.ErrorCodes.FailedToUpsertVectors, $"Failed to upsert, check if the target collection exists.");
```

Comment:
Could you set HttpRequestException as the inner exception in VectorDbException?

Reply:
done